### ISSUE TITLE
Difficulties to connect from Nextcloud

### ISSUE THREAD

**Issue by bepolymathe** [200ok-ch/organice#96]
Hello,

I'm having a little trouble changing the CORS header parameters only for a particular site... the occurrence https://organice.200ok.ch/

I don't want to take any security risks.

**Comment by munen** [comment]
Hi @bepolymathe 

TL;DR: There's no way around that when you want to use organice.200ok.ch with webdav. This is how the web works(;

Longer version: Since organice is a front-end application, it will login with JavaScript from within the browser - in turn the Cross-Origin Resource Sharing (CORS) headers must be set appropriately. If they are not set, you will not be able to login to your service from a browser. Alternatively, if you’re using a server like Apache or Nginx, you can simply get around CORS by hosting organice on the same domain as your service.

Please note, that when your back-end does not set the correct CORS headers, organice cannot show you a really semantic error message on that. The reason is that browsers hide this information from JavaScript. You will simply get a network error.

If you don't want to set CORS headers to allow organice.200ok.ch, but want to use webdav, you can host organice yourself: https://github.com/200ok-ch/organice/#deployment

**Comment by bepolymathe** [comment]
Hello,

Thank you for that answer. My problem is as follows. I use a Nextcloud instance in an external host that does not want to change the CORS headers to preserve the security of the whole (which I can understand) and that does not plan to host an organice instance (I am the only requester;-)). 

So I think I'm stuck at this point. 
Have a good day.

**Comment by rotlaus** [comment]
Hi,

> in turn the Cross-Origin Resource Sharing (CORS) headers must be set appropriately.
Is there somewhere explained how i can set those Headers in an existing Nextcloud Installation?

Thanks for your help.

**Comment by munen** [comment]
@rotlaus There's documentation on WebDAV in the readme of organice: https://github.com/200ok-ch/organice#webdav

There is also documentation on the required headers and a sample configuration for Apache.

**Comment by munen** [comment]
@bepolymathe  

> So I think I'm stuck at this point.

You're not really stuck. You could use Dropbox, Google Drive, or set up your own webdav server. There's documentation on how to do it for Apache in the readme: https://github.com/200ok-ch/organice#webdav

In any case, good luck!^^

**Comment by agenbite** [comment]
Has anyone managed to configure nextcloud's webdav for use with organice? I'd be very interested in their solution...

**Comment by munen** [comment]
@agenbite Here's `contrib` documentation https://github.com/200ok-ch/organice/tree/master/contrib/organice-nextcloud-nginx
